compiler/plugins/input/StableHLO/stablehlo-iree/Conversion/StableHLOCustomCalls.cpp

```diff
@@ -221,6 +310,242 @@ struct HouseholderReflectorRewriter final
   }
 };
 
+llvm::SmallVector<OpFoldResult> makeIndexArray(llvm::SmallVector<int64_t> idx,
+                                               ImplicitLocOpBuilder b) {
+  llvm::SmallVector<OpFoldResult> res;
+  res.reserve(idx.size());
+  for (auto i : idx) {
+    res.push_back(b.getIndexAttr(i));
+  }
+
+  return res;
+}
+
+Value extractSlice(Value v, llvm::ArrayRef<OpFoldResult> offsets,
+                   llvm::ArrayRef<int64_t> sizes, llvm::ArrayRef<int64_t> dims,
+                   ImplicitLocOpBuilder b) {
+  assert(offsets.size() == dims.size());
+  assert(sizes.size() == dims.size());
+
+  auto vTy = cast<ShapedType>(v.getType());
+  auto rank = vTy.getRank();
+
+  llvm::SmallVector<OpFoldResult> vOffsets(rank, b.getIndexAttr(0));
+  llvm::SmallVector<OpFoldResult> vStrides(rank, b.getIndexAttr(1));
+
+  llvm::SmallVector<int64_t> shape(vTy.getShape());
+  for (int i = 0; i < dims.size(); ++i) {
```

Comment:
How would this one look with an llvm::zip with 3 operands? (dims, sizes and offsets).